src/DataProtection/DataProtection/src/TypeForwardingActivator.cs

```diff
@@ -64,6 +62,27 @@ internal object CreateInstance(Type expectedBaseType, string originalTypeName, o
         }
 
         protected string RemoveVersionFromAssemblyName(string forwardedTypeName)
-            => _versionPattern.Replace(forwardedTypeName, "");
+        {
+            // Type, Assembly, Version={Version}, Culture={Culture}, PublicKeyToken={Token}
+
+            var versionStartIndex = forwardedTypeName.IndexOf("Version=", StringComparison.Ordinal);
+            while (versionStartIndex != -1)
```

Comment:
Are there legitimate scenarios where multiple version strings may appear in the assembly name? If it's test specific, I'd much rather we remove the test than change this code.

Reply:
@GrabYourPitchforks is probably the only one who can comment accurately.  If I were to guess, generics probably cause this issue?

Reply:
From the test cases, looks like its generics:

```
        [InlineData(typeof(GenericType<GenericType<ClassWithParameterlessCtor>>))]
        [InlineData(typeof(GenericType<ClassWithParameterlessCtor>))]
        [InlineData(typeof(GenericType<GenericType<string>>))]
        [InlineData(typeof(GenericType<GenericType<string, string>>))]
```